lib/ecto/query/planner.ex

```diff
@@ -891,11 +891,11 @@ defmodule Ecto.Query.Planner do
       end
 
       case find_source_expr(query, child_ix) do
-        %JoinExpr{qual: qual} when qual in [:inner, :left, :inner_lateral, :left_lateral] ->
+        %JoinExpr{qual: qual} when qual in [:inner, :left, :inner_lateral, :left_lateral, :cross_lateral] ->
```

Comment:
```suggestion
        %JoinExpr{qual: qual} when qual in [:inner, :left, :inner_lateral, :left_lateral] ->
```
I don't know if this change is right. I think the point of this clause is that association joins need to be on the appropriate key. If `:cross` is not allowed here I don't believe `:cross_lateral` should be